src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2924,27 +2924,140 @@ LogicalResult ONNXSplitV11Op::inferShapes(
 //===----------------------------------------------------------------------===//
 
 LogicalResult ONNXSplitToSequenceOp::verify() {
-  ONNXSplitToSequenceOpAdaptor operandAdaptor(*this);
-  Value input = operandAdaptor.input();
-  if (!hasShapeAndRank(input))
+  Value inputValue = input();
+  if (!hasShapeAndRank(inputValue))
     return success(); // Won't be able to do any checking at this stage.
 
-  auto inputType = input.getType().cast<ShapedType>();
-  int64_t inputRank = inputType.getShape().size();
-  int64_t axisIndex = axis();
+  auto inputType = inputValue.getType().cast<ShapedType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+  int64_t inputRank = inputShape.size();
 
+  int64_t axisIndex = axis();
   // axis attribute must be in the range [-r,r-1], where r = rank(input).
   if (axisIndex < -inputRank || axisIndex >= inputRank)
     return onnx_mlir::Diagnostic::emitAttributeOutOfRangeError(
         *this->getOperation(), "axis", axisIndex,
         onnx_mlir::Diagnostic::Range<int64_t>(-inputRank, inputRank - 1));
+  if (axisIndex < 0)
+    axisIndex += inputRank;
+
+  Value splitValue = split();
+  if (splitValue.getType().isa<NoneType>()) {
```

Comment:
Use utility function isFromNone(Value) to check whether the input split is provided or not. The utility can cover different ways of representing None in onnx-mlir.

Reply:
done

Reply:
as an aside, I wonder if it would work to simplify the implementation of isFromNone() to just check if the type is none

in principle, any value with NoneType should mean that the value will be None at runtime

I wonder if the current implementation of isFromNone() wants to deal with a situation where an input has NoneType as a placeholder until shape inference fills in a proper type?